### PR TITLE
Pagination

### PR DESCRIPTION
**Pull request by DenisGurskiy**
https://DenisGurskiy.github.io/react_pagination/

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,18 +1,29 @@
...
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState('5');
```
Comment (Viktor-Kost) [inline src/App.tsx:10]: perPage should be a number. This way, you don't have to cast it to a number every time you use it.

File: src/components/Pagination/Pagination.tsx [hunk 2]
```diff
@@ -1 +1,81 @@
...
+        <a
+          data-cy="prevLink"
+          className="page-link"
+          href="#prev"
+          aria-disabled={currentPage === 1}
+          onClick={() => {
+            if (currentPage > 1) {
+              onPageChange(currentPage - 1);
+            }
+          }}
```
Comment (Viktor-Kost) [inline src/components/Pagination/Pagination.tsx:37]: move to a handler

File: src/components/Pagination/Pagination.tsx [hunk 3]
```diff
@@ -1 +1,81 @@
...
+        >
+          <a
+            data-cy="pageLink"
+            className="page-link"
+            href={`#${page}`}
+            onClick={() => {
+              if (page !== currentPage) {
+                onPageChange(page);
+              }
+            }}
```
Comment (Viktor-Kost) [inline src/components/Pagination/Pagination.tsx:55]: move to a handler

File: src/components/Pagination/Pagination.tsx [hunk 4]
```diff
@@ -1 +1,81 @@
...
+        <a
+          data-cy="nextLink"
+          className="page-link"
+          href="#next"
+          aria-disabled={currentPage === count}
+          onClick={() => {
+            if (currentPage < count) {
+              onPageChange(currentPage + 1);
+            }
+          }}
```
Comment (Viktor-Kost) [inline src/components/Pagination/Pagination.tsx:74]: to a handler

**Review by Viktor-Kost — CHANGES_REQUESTED** [review]

File: src/components/Pagination/Pagination.tsx [hunk 5]
```diff
@@ -1 +1,81 @@
...
+        >
+          »
+        </a>
+      </li>
+    </ul>
+  );
```
[the remark this answers is not in the document]
Comment (DenisGurskiy) [reply src/components/Pagination/Pagination.tsx:80]: In the task only <ul className="Pagination"> should be in component Pagination.

**Review by nazarmatsevych — APPROVED** [review]
👍

**Pull request closed without merging**